lineapy/utils/analytics/usage_tracking.py

```diff
@@ -33,6 +35,32 @@ def _py_version():
     )
 
 
+@lru_cache(maxsize=1)
+def _runtime():
+    if get_ipython() is None:
+        runtime = "non-ipython"
+    else:
+        envars: str = ";".join(list(os.environ))
```

Comment:
I thought you need to parse `get_ipython()` to see what kind you are in? 

Now it seems like you are using `os.environ`? Seems like a different strategy?

I would include some block comment on how this works.

Reply:
Yeah, I chose a different strategy as `get_ipython()` does not really tell you what kind you are in. For instance, it returns `ipykernel.zmqshell.ZMQInteractiveShell` object in both Databricks and local Jupyter; and does not seem to contain attribute(s) to distinguish the two. Hence, what we saw for Colab (i.e., `google.colab._shell.Shell`) appears to be a rather exception.

The new strategy is to collect and concatenate all env var names to see if this concatenated string contains a pattern unique to each runtime (e.g., "COLAB_" for Colab, `DATABRICKS_` for Databricks). For runtime types considered here (Colab, Binder, Databricks, Deepnote), this happens to be the case. Let me know if you have concerns about this strategy. Regardless, I will shortly add this explanation to the code.

Reply:
Ok, will do!